main.py

```diff
@@ -161,27 +181,30 @@ def scrape(self, site, depth=0):
                 if not headers:
                     headers = self.get_headers(full_href)
                     self.site_data[full_href]["headers"] = headers
-                new = False
+                else:
+                    current_etag = headers.get("etag")
+                    previous_etag = self.site_data[full_href].get("etag")
+                    if previous_etag != current_etag and current_etag is not None:
```

Comment:
You need to set `new` to `False` for both the `not headers` case and the case where the etags match or don't exist, otherwise it just keeps the value from the last loop iteration.  This gets confusing with all the nesting, may want to refactor this loop.

Reply:
Since I'll specify new=False at the top of each loop, then it will be by default false unless it matches one of the true conditions. It won't borrow from the last loop iteration anymore